service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,133 @@
         </actions>
     </service>
 
+    <!-- ========== Preparing Shipment Map for Label Service ========== -->
+
+    <service verb="create" noun="ShipmentRequestMap">
+        <in-parameters>
+            <parameter name="shipmentId"/>
+            <parameter name="shipmentRouteSegmentSeqId"/>
+            <parameter name="packageRouteSegList" type="List" required="true"/>
+            <parameter name="routeSegment" type="EntityValue" required="true"/>
+            <parameter name="shipment" type="EntityValue" required="true"/>
+            <parameter name="shipmentMethodEnumId"/>
+            <parameter name="shippingGatewayConfigId" required="true"/>
+        </in-parameters>
+        <out-parameters>
+            <parameter name="shipmentRequestMap" type="Map"/>
+        </out-parameters>
+        <actions>
+
+            <!-- Set shippingGatewayConfigId given in input else take value stored in routeSegment entity -->
+            <set field="shippingGatewayConfigId" from="shippingGatewayConfigId?:routeSegment.shippingGatewayConfigId"/>
+
+            <set field="carrierPartyId" from="carrierPartyId ?: routeSegment.carrierPartyId"/>
+            <if condition="!carrierPartyId || carrierPartyId == '_NA_'">
+                <return message="Carrier not set on ${shipmentId}:${shipmentRouteSegmentSeqId}"/>
+            </if>
+
+            <!-- Get ShipmentMethodTypeEnumId from the input or from routeSegment entity-->
+            <set field="shipmentMethodEnumId" from="shipmentMethodEnumId?:routeSegment.shipmentMethodEnumId"/>
+            <if condition="!shipmentMethodEnumId">
+                <return type="danger" message="In get#ShippingLabel Service, no shipmentMethod available."/>
+            </if>
+
+            <set field="hasIssues" from="false"/>
+            <set field="fromPartyId" from="shipment.fromPartyId"/>
+            <if condition="!fromPartyId">
+                <set field="hasIssues" from="true"/>
+                <message type="danger">From party not set on shipment ${shipmentId}</message>
+            </if>
+            <set field="toPartyId" from="shipment.toPartyId"/>
+            <if condition="!toPartyId">
+                <set field="hasIssues" from="true"/>
+                <message type="danger">To party not set on shipment ${shipmentId}</message>
+            </if>
+
+            <!--address_to-->
+            <set field="destPostalContactMech" from="routeSegment.destPostalContactMechId"/>
+            <!-- If ShipmentRouteSegment Entity has no value of destPostalContactMechId, find contactMechId from PartyContactMech -->
+            <if condition="!originPostalContactMech">
```

Comment:
i think there is a typo here in the usage of originPostalContactMech instead of destPostalContactMech.